exonum-light-client/src/main/java/com/exonum/client/response/TransactionResponse.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.client.response;
+
+import com.exonum.binding.common.message.TransactionMessage;
+import com.exonum.binding.common.transaction.TransactionLocation;
+import com.exonum.binding.common.transaction.TransactionResult;
+import lombok.NonNull;
+import lombok.Value;
+
+@Value
+public class TransactionResponse {
+  /**
+   * Current status of the transaction.
+   */
+  @NonNull
+  TransactionStatus status;
+  /**
+   * Transaction message.
+   */
+  @NonNull
+  TransactionMessage message;
+  /**
+   * Transaction execution result; {@code null} - for in-pool transactions.
```

Comment:
What if we define the getter that _throws_ `IllegalStateE` instead? That will roughly align the behaviour of this class with `Optional`.

Alternatively we can return `Optional<TransactionResult|TransactionLocation>`, but in that case the clients who have checked the status would be penalized with an additional `ifPresent/isPresent + get`. 

What do you think would be more convenient:
 1. `null`
 1. `Optional`
 1. not-`null` with a required check of the status?

Reply:
By going with the third approach you will also need to check the status before calling getter. Only one thing that can be considered as a proc (comparing with the first approach) is to have some error description instead of NPE

Reply:
That's not one thing:
1. You get an error early, rather than in undefined time range. And yes, it has an error description.
1. There is a clear connection between the tx status and possible absence of a value — you don't have to check both values for `null`s "just in case" if you have checked if the transaction is committed.